rocketpy/mathutils/function.py

```diff
@@ -1927,13 +1941,22 @@ def __neg__(self):
                 self.__extrapolation__,
             )
         else:
-            return Function(
-                lambda x: -self.source(x),
-                self.__inputs__,
-                self.__outputs__,
-                self.__interpolation__,
-                self.__extrapolation__,
-            )
+            if self.__dom_dim__ == 1:
+                return Function(
+                    lambda x: -self.source(x),
+                    self.__inputs__,
+                    self.__outputs__,
+                )
+            else:
+                param_names = [f"x{i}" for i in range(len(self.__inputs__))]
```

Comment:
Does it work for any N-d function? Not only 2 or 3D?

Reply:
Correct. For instance:

```bash
>>> a=Function(lambda x,y,z,w,m,n,k,q: x+y)
>>> b = -a
>>> a(1,5,3,4,5,6,7,8)
6
>>> b(1,5,3,4,5,6,7,8)
-6
>>> a
'Function from R8 to R1 : (x, y, z, w, m, n, k, q) → (Scalar)'
>>> b
'Function from R8 to R1 : (x, y, z, w, m, n, k, q) → (Scalar)'
```